src/es-module-shims.js

```diff
@@ -229,17 +209,39 @@ async function processScripts () {
     }
     else {
       importMapPromise = importMapPromise.then(async () =>
-        importMap = resolveAndComposeImportMap(script.src ? await (await fetch(script.src)).json() : JSON.parse(script.innerHTML), script.src || pageBaseUrl, importMap)
+        importMap = importShim.resolveImportMap(script.src ? await (await importShim.fetch(script.src)).json() : JSON.parse(script.innerHTML), script.src || pageBaseUrl, importMap)
       );
     }
     script.ep = true;
   }
 }
 
-function resolve (id, parentUrl) {
+function resolve (id, parentUrl, importMap) {
   return resolveImportMap(importMap, resolveIfNotPlainOrUrl(id, parentUrl) || id, parentUrl) || throwUnresolved(id, parentUrl);
 }
 
 function throwUnresolved (id, parentUrl) {
   throw Error("Unable to resolve specifier '" + id + (parentUrl ? "' from " + parentUrl : "'"));
 }
+
+const importShimOptions = self.importShimOptions || {};
```

Comment:
I would prefer not using globals in an API. Is it absolutely necessary to your use case?

Reply:
the current default behavior immediately calls `importShim.load()`, giving no way to customize the shim before it starts processing module/importmap shims on the DOM. an alternative would be to introduce a breaking change and remove the immediate `load` call in favor of the consumer having to manually call it.

Reply:
Another alternative might be to run the initial `load` call on a zero timeout so that a sync script could still alter options before it starts.

Reply:
I can give that a shot, as for overriding should there be `importShim.config` function to apply? or should the consumer just directly mutate the global?

Reply:
Mutation is fine.